app/jobs/email_delivery_jobs/base.rb

```diff
@@ -0,0 +1,82 @@
+module EmailDeliveryJobs
+  class Base < ActiveJob::Base
+
+    #
+    # Send a single email to a recipient informing them about a petition that they have signed
+    # Implemented as a custom job rather than using action mailers #deliver_later so we can do
+    # extra checking before sending the email
+    #
+
+    queue_as :default
+
+    def perform(signature:, timestamp_name:, petition:,
+      requested_at_as_string:, mailer: PetitionMailer.name, logger: nil)
+
+      @mailer = mailer.constantize
+      @signature = signature
+      @petition = petition
+      @requested_at = requested_at_as_string.in_time_zone
+      @timestamp_name = timestamp_name
+      @logger = logger
+
+      if can_send_email?
+        send_email
+        record_email_sent
+      end
+    end
+
+    private
+
+    attr_reader :mailer, :signature, :timestamp_name, :petition, :requested_at
+
+    def can_send_email?
+      petition_has_not_been_updated? && email_not_previously_sent?
+    end
+
+    def send_email
+      create_email.deliver_now
+
+      rescue Errno::ECONNREFUSED, Errno::ETIMEDOUT, Net::SMTPError, Timeout::Error => e
```

Comment:
Have you looked at the AppSignal list of errors we've received - we may want to adjust our handling based on what error we receive.